File: mmr/src/main/java/org/halo/thallo/mmr/core/model/Attribute.java
package org.halo.thallo.mmr.core.model;


import org.halo.thallo.mmr.core.model.op.Operation;

/**
 * Created by dell01 on 2017/8/14.
 */
public interface Attribute extends Model {

    Object getValue();

    boolean isInsertable();

    void setValue(Object val);

    ValueType getValueType();

    void setValueType(ValueType valueType);

    /**
     * 对属性的操作
     * @return
     */
    Operation getOperation();
}

File: mmr/src/main/kotlin/org/halo/thallo/mmr/core/model/Attribute.kt
package org.halo.thallo.mmr.core.model


import org.halo.thallo.mmr.core.model.op.Operation

/**
 * Created by dell01 on 2017/8/14.
 */
interface Attribute : Model {

    var value: Any?

    /**
     * 数据类型
     */
    var type: ValueType

    val length: Int

    var isPrimary: Boolean

    var valueProvider: ValueProvider

    val updateAble: Boolean

    val insertable: Boolean
}
